### ISSUE TITLE
Support Gitea Package Registries (e.g. for Debian, RPM, Alpine,...)

### ISSUE THREAD

**Issue by CMiksche** [goreleaser/goreleaser#4068]
### Is your feature request related to a problem? Please describe.

While using Gitea, packages and binaries are currently just being uploaded as assets to releases. 

But Gitea now provides its own package registries. In the current development version, there is e.g. a registry for Debian, RPM, Alpine and many others.

Instead of uploading it as a Asset, we could upload it directly to the Gitea Package Registry which would make distribution for the users easier.

In my use case, I have multiple packages which I want to distribute to multiple servers via the Gitea Package Registry. I can already do that but would need to write my own code for adding it to the Gitea Registry.

### Describe the solution you'd like

We already have the Gitea Token, Repository and Organisation or User as Env Vars in this project, for supporting the Gitea Package registry, just a few more API calls would have to be added.

E.g. to publish a package:

```
curl --user your_username:your_password_or_token \
     --upload-file path/to/file.deb \
     https://gitea.example.com/api/packages/testuser/debian/pool/bionic/main/upload
```

(See: https://docs.gitea.com/next/usage/packages/packages/debian)

### Describe alternatives you've considered

The current alternative would be that every user creates his own implementation. My plan would be to make the API-Call myself in the Pipeline and upload the package there.

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/caarlos0/) of [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
interesting! seems very doable, just don't have the bandwidth for this release, will try it for the next one.

contributions also welcome :)

**Comment by tashian** [comment]
We've had a lot of folks asking us for proper package repos and this could be a good approach!
+1

**Comment by lng2020** [comment]
I think `gitea/go-sdk`  should support the uploading package sdk first, otherwise, there is no way for goreleaser to support this feature. Please let me know if I'm missing something or if there is another workaround.

**Comment by caarlos0** [comment]
maybe @jolheiser have more info?

**Comment by lunny** [comment]
> I think `gitea/go-sdk` should support the uploading package sdk first, otherwise, there is no way for goreleaser to support this feature. Please let me know if I'm missing something or if there is another workaround.

I don't think this need Gitea's support except generic packages. all other packages have their own official tools to upload/download packages.

**Comment by yonas** [comment]
@lng2020 You can use their HTTP API:

```sh
curl --user your_username:your_password_or_token \
     --upload-file path/to/file.deb \
     https://gitea.example.com/api/packages/testuser/debian/pool/bionic/main/upload
```

**Comment by caarlos0** [comment]
PS: gonna convert this into a discussion until its better defined.
